src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ReferencesPage/ImportedNamespacesValueProvider.cs

```diff
@@ -40,64 +41,57 @@ private async Task<ImmutableArray<string>> GetSelectedImportsAsync()
     
     private async Task<string> GetSelectedImportStringAsync()
     {
-        StringBuilder sb = new StringBuilder();
         string projectName = Path.GetFileNameWithoutExtension(_configuredProject.UnconfiguredProject.FullPath);
         bool containsProjectName = false;
 
-        ImmutableArray<string> existingImports = await GetSelectedImportsAsync();
-        
-        foreach (string? import in existingImports)
-        {
-            if (!string.IsNullOrEmpty(import))
-            {
-                if (string.Equals(import, projectName))
-                {
-                    containsProjectName = true;
-                }
-                sb.Append(import).Append(';');
-            }
-        }
+        ImmutableArray<(string Value, bool IsImported)> existingImports = await GetSelectedImportsAsync();
+
+        var selectedImports = existingImports
+                .Where(pair => !string.IsNullOrEmpty(pair.Value))
+                .Select(pair => (Key: pair.Value, Value: pair.IsImported.ToString())).ToList();
 
         if (!containsProjectName)
```

Comment:
Looks like `containsProjectName` will always be false as it is only assigned at the definition.

Reply:
thank you for the catch